### PR TITLE
[DOC] Edits to connector/task restart operations doc

### PR DESCRIPTION
**Pull request by laidan6000**
### Type of change

Documentation

### Description

Guide(s) updated:
- *Using Strimzi*
- *Deploying and Upgrading Strimzi*

This pull request makes minor edits to the content that was added in #4114 for restarting Kafka Connect and Mirror Maker 2.0 connectors via annotations.

The changes to the procedures are mainly related to Asciidoc formatting and adding some cross-references.

I also made some changes to the explanations in `con-deploy-kafka-connect-managing-connectors.adoc`. Please check that this make sense.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [X] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [X] Reference relevant issue(s) and close them after merging - NO ISSUES FOUND
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: documentation/modules/proc-manual-restart-mirrormaker2-connector.adoc [hunk 1]
```diff
@@ -14,17 +13,32 @@ This procedure describes how to manually trigger a restart of a Kafka MirrorMake
...
-[source,shell,subs=+quotes]
-kubectl describe KafkaMirrorMaker2 _KafkaMirrorMaker2-name_
-
-. To restart the connector, annotate the `KafkaMirrorMaker2` resource in Kubernetes. For example, using `kubectl annotate` to restart the connector named `my-source->my-target.MirrorSourceConnector`:
-[source,shell,subs=+quotes]
-kubectl annotate KafkaMirrorMaker2 _KafkaMirrorMaker2-name_ "strimzi.io/restart-connector=my-source->my-target.MirrorSourceConnector"
++
+[source,shell,subs="+quotes"]
+----
+kubectl describe KafkaMirrorMaker2 _KAFKAMIRRORMAKER-2-NAME_
```
Comment (laidan6000) [inline documentation/modules/proc-manual-restart-mirrormaker2-connector.adoc:26]: Would it be helpful to show the output of this command?

Comment (scholzj) [reply documentation/modules/proc-manual-restart-mirrormaker2-connector.adoc:26]: I don't think that is necessary.

**Review by ppatierno — APPROVED** [review]

File: documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc [hunk 2]
```diff
@@ -14,19 +13,33 @@ This procedure describes how to manually trigger a restart of a Kafka MirrorMake
...
-[source,shell,subs=+quotes]
-kubectl describe KafkaMirrorMaker2 _KafkaMirrorMaker2-name_
-
-. To restart the connector task, annotate the `KafkaMirrorMaker2` resource in Kubernetes.
-For example, using `kubectl annotate` to restart task 0 of the connector named `my-source->my-target.MirrorSourceConnector`:
-[source,shell,subs=+quotes]
-kubectl annotate KafkaMirrorMaker2 _KafkaMirrorMaker2-name_ "strimzi.io/restart-connector-task=my-source->my-target.MirrorSourceConnector:0"
+----
+
+. Find the name of the Kafka MirrorMaker 2.0 connector and the ID of the task to be restarted from the `KafkaMirrorMaker2` custom resource.
```
Comment (PaulRMellor) [inline documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc:22]:
```suggestion
. Find the name of the Kafka MirrorMaker 2.0 connector, and the ID of the task to be restarted from the `KafkaMirrorMaker2` custom resource. 
```

File: documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc [hunk 3]
```diff
@@ -14,19 +13,33 @@ This procedure describes how to manually trigger a restart of a Kafka MirrorMake
...
+. Find the name of the Kafka MirrorMaker 2.0 connector and the ID of the task to be restarted from the `KafkaMirrorMaker2` custom resource. 
+Task IDs are non-negative integers, starting from 0.
++
+[source,shell,subs="+quotes"]
+----
+kubectl describe KafkaMirrorMaker2 _KAFKAMIRRORMAKER-2-NAME_
+----
+
+. To restart the connector task, annotate the `KafkaMirrorMaker2` resource in Kubernetes. 
+For example, using `kubectl annotate` to restart task 0 of the connector named `+my-source->my-target.MirrorSourceConnector+`:
```
Comment (PaulRMellor) [inline documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc:31]:
```suggestion
In this example, `kubectl annotate` is used to restart task _0_ of a connector named `+my-source->my-target.MirrorSourceConnector+`:
```

Comment (PaulRMellor) [reply documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc:31]: The _->_ part of the name shows as arrow in the strimzi doc

File: documentation/modules/proc-manual-restart-mirrormaker2-connector.adoc [hunk 4]
```diff
@@ -14,17 +13,32 @@ This procedure describes how to manually trigger a restart of a Kafka MirrorMake
...
+----
+
+. To restart the connector, annotate the `KafkaMirrorMaker2` resource in Kubernetes. 
+For example, using `kubectl annotate` to restart the connector named `+my-source->my-target.MirrorSourceConnector+`:
```
Comment (PaulRMellor) [inline documentation/modules/proc-manual-restart-mirrormaker2-connector.adoc:30]:
```suggestion
In this example, `kubectl annotate` restarts a connector named `+my-source->my-target.MirrorSourceConnector+`:
```

**Review by PaulRMellor — APPROVED** [review]
LGTM - couple of suggestions/observations

File: documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc (hunk 3 above)
Comment (laidan6000) [reply documentation/modules/proc-manual-restart-mirrormaker2-connector-task.adoc:31]: After some digging in the Asciidoctor user manual, I added the `+` symbols.

[Escape unconstrained formatting marks](https://docs.asciidoctor.org/asciidoc/latest/text/troubleshoot-unconstrained-formatting/#escape-unconstrained)

"Everything between the plus symbols is escaped from interpolation (attribute references, text formatting, etc.)"

**Pull request merged**